web/settings/settings.go

```diff
@@ -253,6 +254,31 @@ func isMovedError(err error) bool {
 	return ok && j.Code == "moved"
 }
 
+func (h *HTTPHandler) UploadAvatar(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, http.MethodPut, consts.Settings); err != nil {
+		return err
+	}
+	header := c.Request().Header
+	size := c.Request().ContentLength
+	if size > 20_000_000 {
```

Comment:
We could have a `MaxFileSize()` method in the `AvatarFS` struct like the one in the `Fs` interface.

Reply:
Maybe this should be an error returned by `CreateAvatar()`?!